go-controller/hybrid-overlay/pkg/controller/node_linux_test.go

```diff
@@ -33,45 +30,99 @@ import (
 )
 
 const (
-	testOVSMAC     string = "11:22:33:44:55:66"
-	testMgmtMAC    string = "06:05:04:03:02:01"
-	testDRMAC      string = "00:00:00:7a:af:04"
-	testNodeSubnet string = "1.2.3.0/24"
-	testNodeIP     string = "1.2.3.3"
+	testMgmtMAC string = "06:05:04:03:02:01"
+
+	thisNodeSubnet string = "1.2.3.0/24"
+	thisNodeDRIP   string = "1.2.3.3"
+	thisNodeDRMAC  string = "22:33:44:55:66:77"
 )
 
+// returns if the two flowCaches are the same
+func compareFlowCache(returnedFlowCache, expectedFlowCache map[string]*flowCacheEntry) bool {
+	if len(returnedFlowCache) != len(expectedFlowCache) {
+		return false
+	}
+
+	for key, entry := range returnedFlowCache {
+		expectedEntry, ok := expectedFlowCache[key]
+		if !ok {
+			klog.Errorf("Unexpected entry %s in nodes flowCache", entry)
+			return false
+		}
+		if !compareFlowCacheEntry(entry, expectedEntry) {
+			klog.Errorf("The test nodes flowCache entry  %s contains unexpected inforation %s", key, entry)
+			return false
+		}
+	}
+	return true
+
+}
+
+// compares two entries in the flow cache
+func compareFlowCacheEntry(returnedEntry, expectedEntry *flowCacheEntry) bool {
```

Comment:
@JacobTanenbaum same here, return a descriptive error and then just pass that through compareFlowCache() to the actual ginkgo compare int he real test function.

Reply:
done